Tests/SwiftParserTest/ExpressionTests.swift

```diff
@@ -1463,12 +1463,53 @@ final class ExpressionTests: ParserTestCase {
     )
   }
 
-  func testNulCharacterInSourceFile() {
+  func testLiteralWithTrailingClosure() {
+    let expectedDiagnostics = [
+      DiagnosticSpec(
+        message: "consecutive statements on a line must be separated by newline or ';'",
+        fixIts: ["insert newline", "insert ';'"]
+      )
+    ]
+
     assertParse(
-      "let a = 1️⃣\u{0}1",
-      diagnostics: [
-        DiagnosticSpec(message: "nul character embedded in middle of file", severity: .warning)
-      ]
+      "_ = true1️⃣ { return true }",
+      diagnostics: expectedDiagnostics,
+      fixedSource: "_ = true\n{ return true }"
+    )
+    assertParse(
+      "_ = nil1️⃣ { return nil }",
+      diagnostics: expectedDiagnostics,
+      fixedSource: "_ = nil\n{ return nil }"
+    )
+    assertParse(
+      "_ = 11️⃣ { return 1 }",
+      diagnostics: expectedDiagnostics,
+      fixedSource: "_ = 1\n{ return 1 }"
+    )
+    assertParse(
+      "_ = 1.01️⃣ { return 1.0 }",
+      diagnostics: expectedDiagnostics,
+      fixedSource: "_ = 1.0\n{ return 1.0 }"
+    )
+    assertParse(
+      #"_ = "foo"1️⃣ { return "foo" }"#,
+      diagnostics: expectedDiagnostics,
+      fixedSource: #"_ = "foo"\#n{ return "foo" }"#
+    )
+    assertParse(
+      #"_ = /foo/1️⃣ { return /foo/ }"#,
+      diagnostics: expectedDiagnostics,
+      fixedSource: #"_ = /foo/\#n{ return /foo/ }"#
```

Comment:
Sure, though the fixedSource in this case is weird anyway since there's really two options 😅 (I assume we're just trying to do both and the second doesn't apply because we already have an edit)